python/ray/data/_internal/execution/interfaces/physical_operator.py

```diff
@@ -65,18 +56,27 @@ def __init__(
     def get_waitable(self) -> StreamingObjectRefGenerator:
         return self._streaming_gen
 
-    def on_waitable_ready(self):
-        # Handle all the available outputs of the streaming generator.
-        while True:
+    def on_data_ready(self, max_bytes_to_read: Optional[int]) -> int:
+        """Callback when data is ready to be read from the streaming generator.
+
+        Args:
+            max_bytes_to_read: The max size in bytes of blocks to read. If None, read
+                all available blocks. This is used for streaming output backpressure.
+        Returns: The total size in bytes of the blocks that have been read.
+        """
+        read_bytes = 0
+        # If max_bytes_to_read is None, we will read all available blocks.
+        # Otherwise, we will read until we reach max_bytes_to_read.
+        while max_bytes_to_read is None or max_bytes_to_read > 0:
```

Comment:
Does this guarantee that we will read at least one block if the block's size is greater than max_bytes_to_read?

Reply:
good question. It doesn't guarantee now. I think I can make it read at least one block when the op buffer is empty.

Reply:
oh, I misunderstood. Actually it already guarantees. I'll add a note here.